rmw_zenoh_cpp/src/detail/rmw_service_data.hpp

```diff
@@ -118,8 +118,8 @@ class ServiceData final : public std::enable_shared_from_this<ServiceData>
 
   // A shared session
   std::shared_ptr<zenoh::Session> sess_;
-  // The keyexpr string.
-  std::string keyexpr_;
+  // The keyexpr.
+  std::optional<zenoh::KeyExpr> keyexpr_;
```

Comment:
Why is this an optional? I don't see any codepaths that check for `std::nullopt`.